### DISCUSSION TITLE
Replace Custom Form Builders?

### DISCUSSION THREAD

**Discussion by ryanwood** [yippee-fun/phlex#276] [Component architecture]
We have a Rails app that is using TailwindCSS. We've been trying to figure out how best to manage the class name complexity. We've tried just using partials and ViewComponent but neither were a great fit for us. I really like the simplicity of Phlex for crafting views/components. 

It gets a little more interesting when we look at forms. We are currently use a custom form builder to abstract away the structural complexity around forms. I'm trying to see how to use Phlex components to wrap form building instead of using a custom builder and dealing with all the ugly template concatenation using content_tags. 

It seems that using `form_with` in Phlex yields the component (self), not the form builder which make it a little challenging. I guess I'm not clear how best to use Phlex to build forms in a Rails app. Any help would be appreciated.

**Comment by ryanwood** [comment]
As a follow up, I see both #132 and #219. My confusion initially was I was calling `f.object.errors.any?` on the builder (which should have been the model but was `nil`) but the `Phlex::Buffer` class has it's own `object` method so it doesn't hit `method_missing` and proxy to the form builder. I guess form support is still in it's infancy and a WIP.

**Reply by joeldrapper** [reply]
This is a bug in our `form_with` adapter. Let's file an issue to fix that.

Ultimately, Phlex will offer a form abstraction to replace form_with. That probably won't be ready for a couple of weeks.

**Reply by joeldrapper** [reply]
I created an issue for this. #281
